packages/mdc-floating-label/README.md

```diff
@@ -0,0 +1,106 @@
+-->
+
+# Floating Label
+
+Floating labels display the type of input a field requires. Every text field and select should have a label. Labels are aligned with the input line and always visible. They can be resting (when a field is inactive and empty) or floating. The label is a text caption or description for the text field.
+
+## Design & API Documentation
+
+<ul class="icon-list">
+  <li class="icon-list-item icon-list-item--spec">
+    <a href="https://material.io/guidelines/components/text-fields.html#text-fields-layout">Material Design guidelines: Text Fields Layout</a>
+  </li>
+</ul>
+
+## Usage
+
+### HTML Structure
+
+```html
+<label class="mdc-floating-label" for="my-text-field-id">Hint text</label>
+```
+
+### Usage within `mdc-text-field`
+
+```html
+<div class="mdc-text-field">
+  <input type="text" id="my-text-field-id" class="mdc-text-field__input">
+  <label class="mdc-floating-label" for="my-text-field-id">Hint text</label>
+  <div class="mdc-text-field__bottom-line"></div>
+</div>
+```
+
+<!-- TODO(mattgoo): add ### Usage within `mdc-select` once select uses mdc-floating-label -->
+
+#### Avoid Dynamic ID Generation
+
+It's also possible to wrap `mdc-text-field__input` within a `<label>` to avoid dynamic id generation:
+
+```html
+<label class="mdc-text-field">
+  <input type="text" class="mdc-text-field__input">
+  <span class="mdc-floating-label">Hint Text</span>
+  <div class="mdc-text-field__bottom-line"></div>
+</label>
+```
+
+> _NOTE_: Only place a `mdc-floating-label` inside of a text field _if you plan on using
+> JavaScript_. Otherwise, the label must go outside of the text-field, as shown below.
+
+#### Single Line, CSS Only
+
+```html
+<label for="text-field-no-js">TextField with no JS:</label>
+<div class="mdc-text-field">
+  <input type="text" id="text-field-no-js" class="mdc-text-field__input" placeholder="Hint text">
+  <div class="mdc-text-field__bottom-line"></div>
+</div>
+```
+
+### CSS Classes
+
+CSS Class | Description
+--- | ---
+`mdc-floating-label` | Mandatory
+`mdc-floating-label--float-above` | Indicates the label is floating above the text field
+`mdc-floating-label--shake` | Shakes the label
+
+### SCSS Mixins
+
+Mixin | Description
+--- | ---
+`mdc-floating-label-ink-color($color)` | Customizes the ink color of the label.
+`mdc-floating-label-fill-color($color)` | Customizes the fill color of the label.
+`mdc-floating-label-shake-keyframes($modifier, $positionY, $positionX, $scale)` | Generates a CSS `@keyframes` at-rule for an invalid label shake. Used in conjunction with the `mdc-floating-label-shake-animation` mixin.
+`mdc-floating-label-shake-animation($modifier)` | Applies shake keyframe animation to label.
+`mdc-floating-label-float-position($positionY, $positionX, $scale)` | Sets position of label when floating.
+
+### `MDCFloatingLabel`
+
+Method Signature | Description
+--- | ---
+`shake(shouldShake: boolean) => void` | Shakes label if shouldShake is true.
```

Comment:
We might want to clarify a few points in the descriptions for `shake()` and `float()`:

1. They also _stop_ shaking/floating if the argument is `false`
2. They proxy straight to the foundation

We might want to rename `styleShake()` to `shake()` and `styleFloat()` to `float()`, and use the same descriptions for both.

E.g.:

```markdown
### `MDCFloatingLabel`

Method Signature | Description
--- | ---
`shake(shouldShake: boolean) => void` | Shakes or stops shaking the label, depending on the value of `shouldShake`. Proxies to the foundation method of the same name.
`float(shouldFloat: boolean) => void` | Floats or docks the label, depending on the value of `shouldFloat`. Proxies to the foundation method of the same name.
`getWidth() => number` | Returns the width of the label element.

...

### `MDCFloatingLabelFoundation`

Method Signature | Description
--- | ---
`shake(shouldShake: boolean) => void` | Shakes or stops shaking the label, depending on the value of `shouldShake`.
`float(shouldFloat: boolean) => void` | Floats or docks the label, depending on the value of `shouldFloat`.
`getWidth() => number` | Returns the width of the label element.
```

Reply:
@acdvorak is renaming these from styleShake to shake and styleFloat to float a breaking change?

Reply:
Isn't this entire package new to this PR? You can't break something that wasn't committed/released yet.

Reply:
True. I did remove the label "package" from textfield. Is that a breaking change?

Reply:
It would be from the perspective of anyone wrapping our components, so it's probably worth mentioning it moved to its own separate package.